src/common/libflux/message.c

```diff
@@ -72,25 +85,30 @@ static int proto_set_type (uint8_t *data, int len, int type)
         return -1;
     switch (type) {
         case FLUX_MSGTYPE_REQUEST:
-            if (proto_set_bigint (data, len, FLUX_NODEID_ANY) < 0)
+            if (proto_set_u32 (data, len, PROTO_IND_NODEID,
+                               FLUX_NODEID_ANY) < 0)
                 return -1;
-            if (proto_set_bigint2 (data, len, FLUX_MATCHTAG_NONE) < 0)
+            if (proto_set_u32 (data, len, PROTO_IND_MATCHTAG,
+                               FLUX_MATCHTAG_NONE) < 0)
                 return -1;
             break;
         case FLUX_MSGTYPE_RESPONSE:
-            if (proto_set_bigint (data, len, 0) < 0)
+            /* N.B. don't clobber matchtag from request on set_type */
+            if (proto_set_u32 (data, len, PROTO_IND_ERRNUM, 0) < 0)
                 return -1;
             break;
         case FLUX_MSGTYPE_EVENT:
-            if (proto_set_bigint (data, len, 0) < 0)
+            if (proto_set_u32 (data, len, PROTO_IND_SEQUENCE, 0) < 0)
                 return -1;
-            if (proto_set_bigint2 (data, len, 0) < 0)
+            if (proto_set_u32 (data, len, PROTO_IND_AUX2, 0) < 0)
                 return -1;
             break;
         case FLUX_MSGTYPE_KEEPALIVE:
-            if (proto_set_bigint (data, len, 0) < 0)
+            if (proto_set_u32 (data, len, PROTO_IND_AUX1, 0) < 0)
```

Comment:
shouldn't do the `PROTO_IND_AUX1` setting?  You've already got `IND_STATUS` and `IND_ERRNUM` below this.

Reply:
Nice catch!  That was a rebase fail from a protocol change I was working on.